apps/web/app/features/stacking/components/choose-rewards-address.tsx

```diff
@@ -7,19 +7,31 @@ import { StackingPoolFormSchema } from '~/features/stacking/utils/stacking-pool-
 import { Input } from '@leather.io/ui';
 
 export function ChooseRewardsAddress() {
-  const { register, getFieldState } = useFormContext<StackingPoolFormSchema>();
-
-  const { isTouched, error } = getFieldState('rewardAddress');
-
-  const showError = isTouched && error;
+  const { control } = useFormContext<StackingPoolFormSchema>();
 
   return (
     <Box>
-      <Input.Root>
-        <Input.Label>Address</Input.Label>
-        <Input.Field id="rewardAddress" {...register('rewardAddress')} />
-      </Input.Root>
-      {showError && <ErrorLabel>{error.message}</ErrorLabel>}
+      <Controller
+        control={control}
+        name="rewardAddress"
+        render={({ field: { onChange, onBlur, value, ref }, fieldState: { invalid, error } }) => (
+          <>
+            <Input.Root>
+              <Input.Label>Address</Input.Label>
+              <Input.Field
```

Comment:
Don't you want to spread the values of render onto the input.field?

Reply:
yes, I tried such approach, but for some reason it doesn't work
in mobile we also use Controller btw